packages/blockchain-fabric-admin/src/LifecycleChannel.ts

```diff
@@ -124,6 +126,49 @@ export class LifecycleChannel {
         return this.submitTransaction(peerNames, ordererName, options, this.COMMIT, requestTimeout);
     }
 
+    public async instantiateOrUpgradeSmartContractDefinition(peerNames: string[], ordererName: string, options: SmartContractDefinitionOptions, fcn: string, args: Array<string>, isUpgrade: boolean, requestTimeout?: number): Promise<void> {
+        const functionName: string = isUpgrade ? this.UPGRADE : this.INSTANTIATE
+        logger.debug('%s - start', functionName);
+
+        const specArgs: Buffer[] = [];
+        const specFcn: string = typeof fcn === 'undefined' ? 'init' : fcn;
+        if (specFcn) {
+            specArgs.push(Buffer.from(specFcn, 'utf8'));
+        }
+        if (!args) {
+            args = [];
+        }
+        for (const arg of args) {
+            specArgs.push(Buffer.from(arg, 'utf8'));
+        }
+
+        const ccSpec: any = {
+            // TODO Golang is defaul in v1 fabric-client, and v1 exension always used default. Should we try to extract language from packaged contracts somehow?
+            type: protos.protos.ChaincodeSpec.Type.GOLANG,
```

Comment:
Any idea how the type is being used? 👀

Reply:
Had a quick look, only see spec being used for name and version. Will need to look at it better later on.
https://github.com/hyperledger/fabric/blob/cae9a63b5dd54adef10e4a27a93e24a87e675f6f/core/scc/lscc/lscc.go#L807